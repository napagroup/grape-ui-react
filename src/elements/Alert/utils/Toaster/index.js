export * from './component';
export * from './props';
export * from './styled';
export * from './styledHelpers';
export * from './utils';
