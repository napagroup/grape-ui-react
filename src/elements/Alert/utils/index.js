export * from './props';
export * from './styledHelpers';
