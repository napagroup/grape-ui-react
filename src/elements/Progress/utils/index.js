export * from './keyframes';
export * from './props';
export * from './styledHelpers';
