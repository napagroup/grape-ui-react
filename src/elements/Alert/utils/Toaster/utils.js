import React from 'react';
import { toast } from 'react-toastify';
import { Alert } from 'src/elements/Alert';
import { removeSomeProps } from 'src/utils/componentHelpers';
import { resolveBoxShadow } from 'src/utils/styledHelpers';

const alertToast = (alertMessage, props = {}) => () => {
  const {
    autoClose,
    bodyClassName,
    className,
    closeButton,
    closeOnClick,
    containerId,
    delay,
    draggable,
    draggablePercent,
    hideProgressBar,
    onClick,
    onClose,
    onOpen,
    pauseOnFocusLoss,
    pauseOnHover,
    position,
    progressClassName,
    progressStyle,
    render,
    role,
    style,
    transition,
  } = props;
  const nativeToastProps = {
    autoClose,
    bodyClassName,
    className,
    closeButton,
    closeOnClick,
    containerId,
    delay,
    draggable,
    draggablePercent,
    hideProgressBar,
    onClick,
    onClose,
    onOpen,
    pauseOnFocusLoss,
    pauseOnHover,
    position,
    progressClassName,
    progressStyle,
    render,
    role,
    style,
    transition,
  };
  return toast(
    <Alert
      alertActionProps={{
        mr: 2,
      }}
      boxShadow={resolveBoxShadow('02')}
      progressProps={{
        animationDuration: `${autoClose}ms`,
        isDeterminate: true,
        loop: false,
      }}
      showProgress={!hideProgressBar && autoClose > 0}
      {...removeSomeProps(props, Object.keys(nativeToastProps))}
    >
      {alertMessage}
    </Alert>, { nativeToastProps }
  );
};

export { alertToast };
