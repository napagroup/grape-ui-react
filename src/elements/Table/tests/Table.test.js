/* eslint-disable react/prop-types */

import React from 'react';
import * as ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom';
import {
  render,
} from '@testing-library/react';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { CodeBlock } from 'src/elements/typography'; // from 'grape-ui-react'
import { Table } from '..';
import { columns, testData } from './utils';

describe('Table - kitchen sink snapshot', () => {
  const pageOptions = [5, 10, 15];
  const pageSize = pageOptions[0];
  const expanderCell = {
  // Make an expander cell
    Cell: ({ row }) => (
    // Use Cell to render an expander for each row.
    // We can use the getToggleRowExpandedProps prop-getter
    // to build the expander.
      <span {...row.getToggleRowExpandedProps()}>
        {row.isExpanded ? 'ðŸ‘‡' : 'ðŸ‘‰'}
      </span>
    ),
    Header: () => null, // No header
    id: 'expander', // It needs an ID
  };
  const renderRowSubComponent = ({ row }) => (
    <CodeBlock code={JSON.stringify({ values: row.values }, null, 2)} />
  );
  const expandableColumns = [expanderCell, ...columns];
  let container;
  beforeEach(() => {
    container = document.createElement('div');
  });
  it('should return table with all components & styling', () => {
    const data = testData;
    const tableBodyProps = {
      tableStriped: {
        even: { bg: 'green' },
        odd: { bg: 'green.light' },
      },
    };
    act(() => {
      ReactDOM.render(
        <ThemeProvider theme={{}}>
          <Table
            columns={expandableColumns}
            data={data}
            enableRowExpansion
            pageOptions={pageOptions}
            pageSize={pageSize}
            renderRowSubComponent={renderRowSubComponent}
            showPagination
            tableBodyProps={tableBodyProps}
          />
        </ThemeProvider>,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });
});

describe('Table - page captions', () => {
  const getCaptionForPage = pageVm => {
    const { pageCount, pageShowCount, pageSize } = pageVm;
    return `Showing ${pageShowCount} of about ${pageCount * pageSize} results`;
  };
  const data = [...testData];
  const pageOptions = [7, 10, 15];
  const pageSize = pageOptions[0];
  const pageCount = Math.ceil(data.length / pageSize, 10);
  it('should return correct page stats on first page', () => {
    const pageIndex = 0;
    const pageShowCount = pageSize;
    const renderUtils = render(
      <ThemeProvider theme={{}}>
        <Table
          columns={columns}
          data={data}
          pageCount={pageCount}
          pageIndex={pageIndex}
          pageOptions={pageOptions}
          pageSize={pageSize}
          showPagination
        />
      </ThemeProvider>
    );
    const { container, getByText } = renderUtils;
    const caption = getCaptionForPage({ pageCount, pageShowCount, pageSize });
    expect(caption).toEqual('Showing 7 of about 203 results');
    expect(container).toContainElement(getByText(caption));
  });
  it('should return correct page stats on last page', () => {
    const pageIndex = Math.ceil(data.length / pageSize) - 1;
    const pageShowCount = data.length % pageSize;
    const renderUtils = render(
      <ThemeProvider theme={{}}>
        <Table
          columns={columns}
          data={data}
          fetchData={() => {}}
          pageCount={pageCount}
          pageIndex={pageIndex}
          pageOptions={pageOptions}
          pageSize={pageSize}
          showPagination
        />
      </ThemeProvider>
    );
    const { container, getByText } = renderUtils;
    const caption = getCaptionForPage({ pageCount, pageShowCount, pageSize });
    expect(pageShowCount).toEqual(4);
    expect(caption).toEqual('Showing 4 of about 203 results');
    expect(container).toContainElement(getByText(caption));
  });
  it('should return correct page stats when data is empty', () => {
    const emptyData = [];
    const emptyPageCount = 0;
    const pageShowCount = 0;
    const renderUtils = render(
      <ThemeProvider theme={{}}>
        <Table
          columns={columns}
          data={emptyData}
          fetchData={() => {}}
        />
      </ThemeProvider>
    );
    const { container, getByText } = renderUtils;
    const caption = getCaptionForPage({ pageCount: emptyPageCount, pageShowCount, pageSize });
    expect(caption).toEqual('Showing 0 of about 0 results');
    expect(container).toContainElement(getByText(caption));
  });
});
