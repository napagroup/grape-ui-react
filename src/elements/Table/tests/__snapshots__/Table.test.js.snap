// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table - kitchen sink snapshot should return table with all components & styling 1`] = `
.c4 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: right;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 0;
  margin-top: 0;
}

.c9 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

.c24 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-left: 4px;
  margin-right: 4px;
}

.c17 {
  position: relative;
  background-color: hsl(0,0%,100%);
  color: inherit;
  padding-bottom: 0;
  padding-top: 4px;
  width: 100%;
}

.c21 {
  position: relative;
  background-color: transparent;
  color: inherit;
  padding-bottom: 0;
  padding-top: 4px;
  width: 100%;
}

.c19 {
  background-color: hsl(0,0%,100%);
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  height: 0.5rem;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 0.8;
  left: 8px;
  position: absolute;
  top: -1px;
  padding-left: 8px;
  padding-right: 8px;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

.c1 {
  box-sizing: border-box;
}

.c5 {
  box-sizing: border-box;
  border-color: hsl(0,0%,90%);
  border-radius: 2px;
  border-style: solid;
  border-width: 1px;
  width: 100%;
  margin-bottom: 4px;
}

.c16 {
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  max-width: 170px;
  margin: 4px;
}

.c20 {
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  max-width: 170px;
  margin-left: 4px;
  margin-right: 4px;
  margin-top: 4px;
  margin-bottom: 4px;
}

.c25 {
  box-sizing: border-box;
  margin-left: 4px;
  margin-right: 4px;
}

.c2 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 4px;
  padding-bottom: 4px;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c23 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 4px;
  margin-bottom: 4px;
}

.c13 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c22 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  border: 1px solid hsl(0,0%,25%);
  border-radius: 2px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-radius: 2px;
  outline: 0;
  width: 100%;
  box-sizing: border-box;
  padding: 1rem;
}

.c22 .grape-ui-select__control,
.c22 .DayPickerInput > input {
  margin: -1rem;
  padding: 1rem;
}

.c22 .grape-ui-select__control + .DayPickerInput-OverlayWrapper,
.c22 .DayPickerInput > input + .DayPickerInput-OverlayWrapper {
  -webkit-transform: translateX(-1rem) translateY(calc(1rem + 2px));
  -ms-transform: translateX(-1rem) translateY(calc(1rem + 2px));
  transform: translateX(-1rem) translateY(calc(1rem + 2px));
}

.c22[disabled] {
  opacity: 0.3;
}

.c22[disabled] ~ * {
  color: rgba(0,0,0,0.3);
}

.c22:focus,
.c22:focus-within {
  border-color: hsl(323.3,84.6%,28%);
  box-shadow: 0 0 0 1px hsl(323.3,84.6%,28%);
}

.c22:focus + label,
.c22:focus-within + label {
  color: hsl(323.3,84.6%,28%);
}

.c22::-webkit-input-placeholder,
.c22 input::-webkit-input-placeholder {
  color: hsl(0,0%,75%);
}

.c22::-moz-placeholder,
.c22 input::-moz-placeholder {
  color: hsl(0,0%,75%);
}

.c22:-ms-input-placeholder,
.c22 input:-ms-input-placeholder {
  color: hsl(0,0%,75%);
}

.c22::placeholder,
.c22 input::placeholder {
  color: hsl(0,0%,75%);
}

.c22 .DayPickerInput > input {
  background: transparent;
  border: 0;
  font: inherit;
  outline: 0;
  width: 100%;
}

.c18 {
  background-color: transparent;
  color: inherit;
  border: 1px solid hsl(0,0%,25%);
  border-radius: 2px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-radius: 2px;
  outline: 0;
  width: 100%;
  box-sizing: border-box;
  padding: 1rem;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

.c18 .grape-ui-select__control,
.c18 .DayPickerInput > input {
  margin: -1rem;
  padding: 1rem;
}

.c18 .grape-ui-select__control + .DayPickerInput-OverlayWrapper,
.c18 .DayPickerInput > input + .DayPickerInput-OverlayWrapper {
  -webkit-transform: translateX(-1rem) translateY(calc(1rem + 2px));
  -ms-transform: translateX(-1rem) translateY(calc(1rem + 2px));
  transform: translateX(-1rem) translateY(calc(1rem + 2px));
}

.c18[disabled] {
  opacity: 0.3;
}

.c18[disabled] ~ * {
  color: rgba(0,0,0,0.3);
}

.c18:focus,
.c18:focus-within {
  border-color: hsl(323.3,84.6%,28%);
  box-shadow: 0 0 0 1px hsl(323.3,84.6%,28%);
}

.c18:focus + label,
.c18:focus-within + label {
  color: hsl(323.3,84.6%,28%);
}

.c18::-webkit-input-placeholder,
.c18 input::-webkit-input-placeholder {
  color: hsl(0,0%,75%);
}

.c18::-moz-placeholder,
.c18 input::-moz-placeholder {
  color: hsl(0,0%,75%);
}

.c18:-ms-input-placeholder,
.c18 input:-ms-input-placeholder {
  color: hsl(0,0%,75%);
}

.c18::placeholder,
.c18 input::placeholder {
  color: hsl(0,0%,75%);
}

.c18 .DayPickerInput > input {
  background: transparent;
  border: 0;
  font: inherit;
  outline: 0;
  width: 100%;
}

.c26 {
  border: 0;
  border-radius: 4px;
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  position: relative;
  margin: 4px;
  padding-bottom: 4px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  text-transform: uppercase;
}

.c26:hover {
  cursor: pointer;
  background-color: hsl(0,0%,90%);
}

.c26:active {
  background-color: hsl(0,0%,100%);
}

.c26[disabled] {
  opacity: 0.4;
  pointer-events: none;
  cursor: not-allowed;
}

.c26 > .c0 {
  border-radius: 4px;
  display: inline-block;
  max-width: 100%;
  vertical-align: top;
}

.c7 {
  border-spacing: 0;
  width: 100%;
}

.c11 {
  background-color: transparent;
  color: inherit;
}

.c11 tr:nth-child(even) {
  background-color: hsl(122.4,39.4%,49.2%);
}

.c11 tr:nth-child(odd) {
  background-color: hsl(122.4,39.4%,69.2%);
}

.c8 {
  border-bottom-color: hsl(0,0%,90%);
  border-bottom-style: solid;
  border-bottom-width: 1px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
  vertical-align: bottom;
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: bold;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

.c8:hover .sort-icon {
  opacity: 0.3;
}

.c12 {
  border-bottom-color: hsl(0,0%,90%);
  border-bottom-style: solid;
  border-bottom-width: 1px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 8px;
  padding-bottom: 8px;
  vertical-align: top;
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

.c12:hover .sort-icon {
  opacity: 0.3;
}

.c6 {
  overflow-x: auto;
}

.c10 {
  opacity: 0;
  position: absolute;
  -webkit-transform: translate(-12px,6px);
  -ms-transform: translate(-12px,6px);
  transform: translate(-12px,6px);
}

@media screen and (min-width:40em) {
  .c24 {
    margin-left: 8px;
    margin-right: 8px;
  }
}

@media screen and (min-width:40em) {
  .c20 {
    margin-left: 8px;
    margin-right: 8px;
  }
}

@media screen and (min-width:40em) {
  .c2 {
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:40em) {
  .c14 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:40em) {
  .c23 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:40em) {
  .c8 {
    padding-left: 16px;
    padding-right: 16px;
  }
}

@media screen and (min-width:40em) {
  .c12 {
    padding-left: 16px;
    padding-right: 16px;
  }
}

<div>
  <div
    class="c0 c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <p
          class="c4"
        >
          Showing 5 of about 200 results
        </p>
      </div>
    </div>
    <div
      class="c0 c5 "
    >
      <div
        class="c0 c1 c6"
      >
        <table
          class="c7"
          role="table"
        >
          <thead
            class=""
          >
            <tr
              class=""
              role="row"
            >
              <th
                class="c8"
                colspan="1"
                role="columnheader"
              >
                <span
                  class="c9"
                  style="position: relative;"
                >
                  
                   
                </span>
              </th>
              <th
                class="c8"
                colspan="2"
                role="columnheader"
                title="test-name-column"
              >
                <span
                  class="c9"
                  style="position: relative;"
                >
                  
                  Name
                </span>
              </th>
              <th
                class="c8"
                colspan="1"
                role="columnheader"
                title="test-info-column"
              >
                <span
                  class="c9"
                  style="position: relative;"
                >
                  
                  Info
                </span>
              </th>
            </tr>
            <tr
              class=""
              role="row"
            >
              <th
                class="c8"
                colspan="1"
                role="columnheader"
              >
                <span
                  class="c9"
                  style="position: relative;"
                >
                  
                </span>
              </th>
              <th
                class="c8"
                colspan="1"
                role="columnheader"
                style="cursor: pointer;"
                title="test-first-name-column"
              >
                <span
                  class="c9"
                  style="position: relative;"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-arrow-up fa-w-14 fa-xs c10 sort-icon"
                    data-icon="arrow-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z"
                      fill="currentColor"
                    />
                  </svg>
                  First Name
                </span>
              </th>
              <th
                class="c8"
                colspan="1"
                role="columnheader"
                style="cursor: pointer;"
                title="test-last-name-column"
              >
                <span
                  class="c9"
                  style="position: relative;"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-arrow-up fa-w-14 fa-xs c10 sort-icon"
                    data-icon="arrow-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z"
                      fill="currentColor"
                    />
                  </svg>
                  Last Name
                </span>
              </th>
              <th
                class="c8"
                colspan="1"
                role="columnheader"
                style="cursor: pointer;"
                title="test-age-column"
              >
                <span
                  class="c9"
                  style="position: relative;"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-arrow-up fa-w-14 fa-xs c10 sort-icon"
                    data-icon="arrow-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z"
                      fill="currentColor"
                    />
                  </svg>
                  Age
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="c11"
            role="rowgroup"
          >
            <tr
              class=""
              role="row"
            >
              <td
                class="c12"
                role="cell"
              >
                <span
                  style="cursor: pointer;"
                  title="Toggle Row Expanded"
                >
                  👉
                </span>
              </td>
              <td
                class="c12"
                role="cell"
              >
                Brian
              </td>
              <td
                class="c12"
                role="cell"
              >
                Davis
              </td>
              <td
                class="c12"
                role="cell"
              >
                18
              </td>
            </tr>
            <tr
              class=""
              role="row"
            >
              <td
                class="c12"
                role="cell"
              >
                <span
                  style="cursor: pointer;"
                  title="Toggle Row Expanded"
                >
                  👉
                </span>
              </td>
              <td
                class="c12"
                role="cell"
              >
                Christine
              </td>
              <td
                class="c12"
                role="cell"
              >
                Garcia
              </td>
              <td
                class="c12"
                role="cell"
              >
                41
              </td>
            </tr>
            <tr
              class=""
              role="row"
            >
              <td
                class="c12"
                role="cell"
              >
                <span
                  style="cursor: pointer;"
                  title="Toggle Row Expanded"
                >
                  👉
                </span>
              </td>
              <td
                class="c12"
                role="cell"
              >
                James
              </td>
              <td
                class="c12"
                role="cell"
              >
                Brown
              </td>
              <td
                class="c12"
                role="cell"
              >
                42
              </td>
            </tr>
            <tr
              class=""
              role="row"
            >
              <td
                class="c12"
                role="cell"
              >
                <span
                  style="cursor: pointer;"
                  title="Toggle Row Expanded"
                >
                  👉
                </span>
              </td>
              <td
                class="c12"
                role="cell"
              >
                Christine
              </td>
              <td
                class="c12"
                role="cell"
              >
                Anderson
              </td>
              <td
                class="c12"
                role="cell"
              >
                20
              </td>
            </tr>
            <tr
              class=""
              role="row"
            >
              <td
                class="c12"
                role="cell"
              >
                <span
                  style="cursor: pointer;"
                  title="Toggle Row Expanded"
                >
                  👉
                </span>
              </td>
              <td
                class="c12"
                role="cell"
              >
                Ann
              </td>
              <td
                class="c12"
                role="cell"
              >
                Patterson
              </td>
              <td
                class="c12"
                role="cell"
              >
                30
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <form
        class="c13"
        role="form"
      >
        <div
          class="c14"
          data-testid="paginator"
        >
          <div
            class="c15"
          >
            <div
              class="c0 c16"
            >
              <div
                class="c17"
              >
                <input
                  class="c18"
                  id="jumpToPage"
                  name="jumpToPage"
                  tabindex="0"
                  type="number"
                  value="1"
                />
                <label
                  class="c19"
                  for="jumpToPage"
                >
                  Jump to Page
                </label>
              </div>
            </div>
            <div
              class="c0 c20"
            >
              <div
                class="c21"
              >
                <div
                  class="c22 css-2b097c-container"
                  id="selectedPageOption"
                >
                  <div
                    class="grape-ui-select__control css-6bcw64-Control"
                  >
                    <div
                      class="grape-ui-select__value-container grape-ui-select__value-container--has-value css-1jw5w5s-ValueContainer"
                    >
                      <div
                        class="grape-ui-select__single-value css-1y5ucjk-SingleValue"
                      >
                        Show 5
                      </div>
                      <div
                        class="css-17yls17-Input"
                      >
                        <div
                          class="grape-ui-select__input"
                          style="display: inline-block;"
                        >
                          <input
                            aria-autocomplete="list"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            id="react-select-2-input"
                            spellcheck="false"
                            style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                          <div
                            style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class="grape-ui-select__indicators css-zededq-IndicatorsContainer"
                    >
                      <span
                        class="grape-ui-select__indicator-separator css-1xjgjl1-IndicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="grape-ui-select__indicator grape-ui-select__dropdown-indicator css-vv1fq9-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-6q0nyr-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <input
                    name="selectedPageOption"
                    type="hidden"
                    value="5"
                  />
                </div>
                <label
                  class="c19"
                  for="selectedPageOption"
                >
                  Rows Per Page
                </label>
              </div>
            </div>
          </div>
          <div
            class="c23"
          >
            <span
              class="c24"
            >
              Page 1 of 40
            </span>
            <div
              class="c0 c25"
            >
              <button
                class="c26"
                data-testid="gotoPreviousPage"
                disabled=""
                role="button"
                type="button"
              >
                <div
                  class="c0 c1"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-angle-left fa-w-8 "
                    data-icon="angle-left"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 256 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </button>
              <button
                class="c26"
                data-testid="gotoNextPage"
                role="button"
                type="button"
              >
                <div
                  class="c0 c1"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-angle-right fa-w-8 "
                    data-icon="angle-right"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 256 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
